zarr/util.py

```diff
@@ -235,19 +235,16 @@ def normalize_fill_value(fill_value, dtype):
         # no fill value
         pass
 
-    elif fill_value == 0 and dtype.kind in 'SV':
-        # special case because 0 used as default, but cannot be used for structured arrays
-        fill_value = b''
+    elif fill_value == 0:
+        # this should be compatible across numpy versions for any array type, including
+        # structured arrays
+        fill_value = np.zeros((), dtype=dtype)[()]
 
     elif dtype.kind == 'U':
-        # special case unicode because of encoding issues on Windows if passed through
-        # numpy
+        # special case unicode because of encoding issues on Windows if passed through numpy
         # https://github.com/alimanfoo/zarr/pull/172#issuecomment-343782713
 
-        if fill_value == 0:
-            fill_value = ''
```

Comment:
Why is this no longer needed?

Reply:
This is now covered by [this branch](https://github.com/zarr-developers/zarr/blob/b113d0aa8295dab550fc54a2d4b16b49d5a61b23/zarr/util.py#L238).

Reply:
Ah sorry. Overlooked that it is handled above generally. Thanks for the pointer. Seems fine.